tests/conftest.py

```diff
@@ -1495,3 +1503,16 @@ def use_ipfs_daemon() -> Generator:
 
     yield
     ipfs_daemon.stop()
+
+
+def pytest_configure(config):
+    """Register the profiling marker to avoid warnings."""
+    config.addinivalue_line("markers", "profiling: filter profiler tests")
+
+
+def pytest_collection_modifyitems(config, items):
+    """Run profiler or non-profiler tests separately."""
+    if config.getoption("--profiling"):
+        items[:] = filter(lambda item: item.get_closest_marker("profiling"), items)
+    else:
+        items[:] = filter(lambda item: not item.get_closest_marker("profiling"), items)
```

Comment:
Here we select which set of tests we will run: profiler or the other ones.

Reply:
The other approach is to just tag the relevant tests with a marker and then explicitly select the marker. Then you limit it to the marked tests (that's how it's done for integration tests for instance).

Reply:
Ok, I'll go that route. That was what I've been trying to do but every post I saw talked about adding pytest_collection_modifyitems. I've just talked to Viraj and I'll create a separate job in the workflow.